js/newArticle.js

```diff
@@ -1,21 +1,43 @@
+var makeNewArticle = {};
+var articlePreview = {};
+var cheese = {};
+
+marked.setOptions({
+  renderer: new marked.Renderer(),
+  gfm: true,
+  tables: true,
+  breaks: true,
+  pedantic: false,
+  sanitize: true,
+  smartLists: true,
+  smartypants: false
+});
+
 $('#write').on('keyup', function(){
-  var $articlePreview = '<h1>'+ $('#article-title').val() + '</h1>';
-  $articlePreview += '<h5>' + $('#article-author').val() + '</h5>';
-  $articlePreview += '<h5><a>' + $('#article-author-url').val() + '</a></h5>';
-  $articlePreview += '<h5>Published on ' + new Date() + '</h5>';
-  $articlePreview += '<h6>Category: ' + $('#article-category').val() + '</h6>';
-  $articlePreview += '```' + $('#article-body').val() + '```';
-  $('#articlesPreview').html(marked($articlePreview));
-
-  $('pre code').addClass('html');
+  articlePreview.title = $('#article-title').val();
+  articlePreview.author = $('#article-author').val();
+  articlePreview.authorUrl = $('#article-author-url').val();
+  articlePreview.publishedOn = new Date();
+  articlePreview.category = $('#article-category').val();
+  articlePreview.body = '<pre><code>' + (marked($('#article-body').val())) + '</code></pre>';
+
+  cheese.articleList = new Article(articlePreview);
+
+  var articleTemplateRun = function () {
+    var theTemplateScript = $('#article-template').html();
+    var theTemplate = Handlebars.compile(theTemplateScript);
+    var theCompiledTemplate = theTemplate(cheese.articleList);
+    $('#articlesPreview').html(theCompiledTemplate);
+  };
+
+  articleTemplateRun();
+
   $('pre code').each(function(i, block) {
+    $('pre code').addClass('html');
     hljs.highlightBlock(block);
   });
 });
 
-var makeNewArticle = {};
-var genJSON;
-
 makeNewArticle.JSON = function() {
```

Comment:
I'm not sure naming this function `JSON` is the best name, since `JSON` is already a builtin JavaScript object. Maybe it's ok, but I'm just not sure.